### PR TITLE
clock: show time and date in a nicer fashion

### PR DESCRIPTION
**Pull request by endian-albin**
- Add leading zeros to single digit numbers
- Remove the seconds component
- Add weekday, day of month and month
- Represent second ticks by toggling the HH:MM separator

### CODE REVIEW

File: hypnos/src/clock.c [hunk 1]
```diff
@@ -70,32 +72,47 @@ void clock_increment_local_time()
...
 	lv_label_set_text(clock_label, clock_label_str);
+	lv_label_set_text(date_label, date_label_str);
 }
 
 void clock_gfx_init()
 {
-	sprintf(clock_label_str, "%d:%d:%d", localtime(&local_time)->tm_hour,
-	       localtime(&local_time)->tm_min,
-	       localtime(&local_time)->tm_sec);
+	char wday[15];
```
Comment (Lukas-Berglund) [inline hypnos/src/clock.c:100]: The char of wday and mon is of size 10 in clock_print_time(). Can size 10 work here as well?

Comment (endian-albin) [reply hypnos/src/clock.c:100]: Actually, I realised that wday and mon are abbreviations (three chars), so I can reduce even more.

**Comment by Lukas-Berglund** [comment]
Looks good. I just had one small question.

File: hypnos/src/clock.c (hunk 1 above)
Comment (endian-albin) [reply hypnos/src/clock.c]: I changed it to `4` in two places it that allowed me to decrease `date_label_str` to size `32`, thanks!

**Pull request merged**